vyper/cli/vyper_compile.py

```diff
@@ -143,65 +154,89 @@ def exc_handler(contract_name: ContractName, exception: Exception) -> None:
     raise exception
 
 
-def get_interface_codes(codes: ContractCodes) -> Any:
-    interface_codes = {}
-    interfaces = {}
+def get_interface_codes(root_path: Path, codes: ContractCodes) -> Any:
+    interface_codes: Dict = {}
+    interfaces: Dict = {}
 
-    for code in codes.values():
-        interface_codes.update(extract_file_interface_imports(code))
+    for file_path, code in codes.items():
+        interfaces[file_path] = {}
+        parent_path = root_path.joinpath(file_path).parent
 
-    if interface_codes:
+        interface_codes = extract_file_interface_imports(code)
         for interface_name, interface_path in interface_codes.items():
```

Comment:
This seems like a major change in control flow logic. Is this supposed to be indented under the previous `for` loop?

Reply:
To be clearer, in the prior setup the `if interface_codes` statement escaped the previous `for` loop. This change sets the `interface_codes` variable directly _however_ it does *not* escape the previous `for` loop anymore.

Reply:
See https://github.com/ethereum/vyper/issues/1367#issuecomment-519484066 -  In the prior setup, the call to `extract_file_interface_imports` happened once per compile which resulted in a single shared interface namespace across all contracts being compiled. Making the call once per contract allows the namespaces to be isolated from each other.